samples/Prometheus/ReverseProxy.Metrics-Promethius.Sample/PrometheusServiceExtensions.cs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+using Yarp.ReverseProxy.Telemetry.Consumption;
+
+namespace Yarp.Sample
+{
+    public static class PrometheusServiceExtensions
+    {
+        public static IServiceCollection AddPrometheusProxyMetrics(this IServiceCollection services)
+        {
+            services.AddProxyTelemetryListener();
+            services.AddSingleton<IProxyMetricsConsumer, PrometheusProxyMetrics>();
+            return services;
+        }
+
+#if NET5_0_OR_GREATER
+        public static IServiceCollection AddPrometheusDnsMetrics(this IServiceCollection services)
+        {
+            services.AddSocketsTelemetryListener();
```

Comment:
```suggestion
            services.AddNameResolutionTelemetryListener();
```

The risk of such typos will go away with #915 where only `AddTelemetryListeners` will be a thing